src/azure-cli-core/azure/cli/core/profiles/_shared.py

```diff
@@ -24,6 +24,12 @@ def __str__(self):
 PROFILE_TYPE = object()
 
 
+class CustomResourceType(object):  # pylint: disable=too-few-public-methods
```

Comment:
Also, what if you have to add an endpoint type in addition to or instead of a resource type?  How would that be supported?

Reply:
That would be part of https://github.com/Azure/azure-cli/issues/4786 which this does not address.

Reply:
Would it not make sense to implement these both at the same time?